cmd/skaffold/app/cmd/config/set.go

```diff
@@ -62,20 +77,16 @@ func setConfigValue(name string, value interface{}) error {
 			}
 		}
 	}
-	if fieldName == "" {
-		return fmt.Errorf("%s is not a valid config field", name)
-	}
-	fieldValue := cfgValue.FieldByName(fieldName)
-
-	fieldType := fieldValue.Type()
-	val := reflect.ValueOf(value)
+	return fieldName
+}
 
-	if fieldType != val.Type() {
-		return fmt.Errorf("%s is not a valid value for field %s", value, fieldName)
+func getNewFieldValue(fieldName string, name string, value string) (reflect.Value, error) {
+	config := ContextConfig{}
+	auxiliaryContents := name + ": " + value
```

Comment:
I'm not sure I understand why you need to do this. What was wrong with just doing a `reflect.ValueOf(newValue)`? Creating a new config and then doing `reflect.ValueOf(config).FieldByName(fieldName)` seems redundant to me, but I might be missing something.

I'm also not crazy about using anything yaml-related in this code: the entire reason I rewrote it to use reflection was to move away from using yaml marshaling/unmarshaling magic to set values in a struct.

Reply:
So the reflection code was working very nicely with just `string` typed values. However, the new option is a `*bool`, so that the old code was trying to set it with a string value. This resulted in a type mismatch in line 73 old version, because `setConfigValue` is always called with actual type `string` for the value. But maybe I'm missing a simpler solution here.

I was very back and forth with another approach where the `LocalCluster` config is also made a `string`, so that the old code would have worked fine here. However, in the end I thought that there might be further settings in the future where such a workaround is even uglier than for `LocalCluster`.